cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1505,6 +1531,9 @@ void writer::impl::write(table_view const& table, std::vector<partition_info> co
         comp_rowgroup_size += ck->compressed_size;
         max_chunk_bfr_size =
           std::max(max_chunk_bfr_size, (size_t)std::max(ck->bfr_size, ck->compressed_size));
+        if (stats_granularity_ == statistics_freq::STATISTICS_COLUMN) {
```

Comment:
This seems odd to me. I would expect a bitmask so you could select what stats you wanted and not a set value for this. Pardon my ignorance, but could you explain a bit how this is working? What if I want column and rowgroup stats, but not page stats?

Reply:
So here's my thinking...first off, parquet deprecated page stats (STATISTICS_PAGE).  It's still in the spec for old readers, but when the column indexes were added, they said one should not encode both page stats and column indexes.  To keep them mutually exclusive I couldn't use a bit field to turn each of them on and off individually.  Now I suppose you could do column indexes without the row group stats, but I can't think of a case where that would be desirable.  As things are coded now, we test for granularity != STATISTICS_NONE to determine whether to do the row group stats.  STATISTICS_PAGE then adds page stats, and now STATISTICS_COLUMN will instead add column indexes.

Reply:
I can buy that.